pkg/goDB/storage/gpfile/buffers.go

```diff
@@ -0,0 +1,108 @@
+package gpfile
+
+import (
+	"fmt"
+	"io"
+	"sync"
+)
+
+type readWriteSeekCloser interface {
+	io.Reader
+	io.Writer
+	io.Seeker
+	io.Closer
+}
+
+// MemPool wraps a standard sync.Pool
+type MemPool struct {
+	sync.Pool
+}
+
+// NewMemPool instantiates a new memory pool that manages bytes slices
+// of a given capacity
+func NewMemPool() *MemPool {
+	return &MemPool{
+		Pool: sync.Pool{
+			New: func() any {
+				return make([]byte, 0)
+			},
+		},
+	}
+}
+
+// Get retrieves a memory element (already performing the type assertion)
+func (p *MemPool) Get() []byte {
+	return p.Pool.Get().([]byte)
+}
+
+// Put returns a memory element to the pool, resetting its size to capacity
+// in the process
+func (p *MemPool) Put(elem []byte) {
+	elem = elem[:cap(elem)]
+
+	// nolint:staticcheck
+	p.Pool.Put(elem)
+}
+
+// MemFile denotes an in-memory abstraction of an underlying file, acting as
+// a buffer (drawing memory from a pool)
+type MemFile struct {
+	data []byte
+	pos  int
+
+	pool *MemPool
+}
+
+// NewMemFile instantiates a new in-memory file buffer
+func NewMemFile(r io.ReadCloser, l int, pool *MemPool) (*MemFile, error) {
```

Comment:
Nitpick: for readability, I'd call `l int` what it is: the `length` of the file.